src/search/landmarks/landmark_count_heuristic.cc

```diff
@@ -242,25 +245,52 @@ void LandmarkCountHeuristic::generate_preferred_operators(
             FactProxy fact_proxy = effect.get_fact();
             LandmarkNode *lm_node = lgraph->get_node(fact_proxy.get_pair());
             if (lm_node && landmark_is_interesting(
-                    state, reached, *lm_node, all_landmarks_reached)) {
+                state, reached, *lm_node, all_landmarks_reached)) {
                 if (lm_node->get_landmark().disjunctive) {
                     preferred_operators_disjunctive.push_back(op_id);
                 } else {
                     preferred_operators_simple.push_back(op_id);
                 }
             }
+            if (lgraph->contains_conjunctive_landmark(fact_proxy.get_pair())) {
+                std::vector<LandmarkNode *> conjunctive_landmarks = lgraph->get_conjunctive_landmarks(fact_proxy.get_pair());
+                for( auto conj_lm : conjunctive_landmarks){
+                    if (landmark_is_interesting(
+                            state, reached, *conj_lm, all_landmarks_reached) ){
+                        preferred_operators_conjunctive.push_back(op_id);
+                        break;
+                    }
+                }
+
+            }
         }
     }
 
     OperatorsProxy operators = task_proxy.get_operators();
-    if (preferred_operators_simple.empty()) {
-        for (OperatorID op_id : preferred_operators_disjunctive) {
-            set_preferred(operators[op_id]);
+    if (use_preferrence_hierarchy){
+        if (!preferred_operators_conjunctive.empty()) {
+            for (OperatorID op_id : preferred_operators_conjunctive) {
+                set_preferred(operators[op_id]);
+            }
+        } else if (!preferred_operators_simple.empty()) {
+            for (OperatorID op_id : preferred_operators_simple) {
+                set_preferred(operators[op_id]);
+            }
+        } else {
+            for (OperatorID op_id : preferred_operators_disjunctive) {
+                set_preferred(operators[op_id]);
+            }
         }
     } else {
+        for (OperatorID op_id: preferred_operators_conjunctive) {
+            set_preferred(operators[op_id]);
+        }
         for (OperatorID op_id : preferred_operators_simple) {
             set_preferred(operators[op_id]);
         }
+        for (OperatorID op_id : preferred_operators_disjunctive) {
+            set_preferred(operators[op_id]);
+        }
     }
```

Comment:
If we pursue the variant without hierarchy (which I hope we can), then this two-stage approach is no longer necessary; `preferred_operators_{conjunctive,simple,disjunctive}`) should no longer exist, and `set_preferred` should instead be called directly.

But this is enough with a code change, together with the other changes that we should make to remove temporary options etc., that I think the experiments should be run again on the version we would like to merge.